Mathlib/FieldTheory/SeparableClosure.lean

```diff
@@ -0,0 +1,374 @@
+    (h : separableClosure E K = ⊥) :
+    separableClosure F K = (separableClosure F E).map (IsScalarTower.toAlgHom F E K) := by
+  refine le_antisymm (fun x hx ↦ ?_) (map_le_of_algHom F E K _)
+  obtain ⟨y, rfl⟩ := mem_bot.1 <| h ▸ (mem_separableClosure_iff E K).2
+    (mem_separableClosure_iff _ _ |>.mp hx |>.map_minpoly E)
+  exact ⟨y, (map_mem_separableClosure_iff _ _ _ <| IsScalarTower.toAlgHom F E K).mp hx, rfl⟩
+
+/-- If `i` is an `F`-algebra isomorphism of `E` and `K`, then `separableClosure F K` is equal to
+the image of `separableClosure F E` under the map `i`. -/
+theorem separableClosure.eq_map_of_algEquiv (i : E ≃ₐ[F] K) :
+    separableClosure F K = (separableClosure F E).map i :=
+  le_antisymm (fun x h ↦ ⟨_, (map_mem_separableClosure_iff F K E i.symm).2 h, i.right_inv x⟩)
+    (map_le_of_algHom F E K i)
+
+/-- If `E` and `K` are isomorphic as `F`-algebras, then `separableClosure F E` and
+`separableClosure F K` are also isomorphic as `F`-algebras. -/
+def separableClosure.algEquivOfAlgEquiv (i : E ≃ₐ[F] K) :
+    separableClosure F E ≃ₐ[F] separableClosure F K :=
+  ((separableClosure F E).intermediateFieldMap i).trans
+    (equivOfEq (eq_map_of_algEquiv F E K i).symm)
+
+/-- The (relative) separable closure of `E / F` is algebraic over `F`. -/
+theorem separableClosure.isAlgebraic : Algebra.IsAlgebraic F (separableClosure F E) :=
+  fun x ↦ isAlgebraic_iff.2 x.2.isIntegral.isAlgebraic
+
+/-- The (relative) separable closure of `E / F` is separable over `F`. -/
+instance separableClosure.isSeparable : IsSeparable F (separableClosure F E) :=
+  ⟨fun x ↦ by simpa only [minpoly_eq] using x.2⟩
+
+/-- An intermediate field of `E / F` is contained in the (relative) separable closure of `E / F`
+if all of its elements are separable over `F`. -/
+theorem le_separableClosure' {L : IntermediateField F E} (hs : ∀ x : L, (minpoly F x).Separable) :
+    L ≤ separableClosure F E := fun x h ↦ by simpa only [minpoly_eq] using hs ⟨x, h⟩
+
+/-- An intermediate field of `E / F` is contained in the (relative) separable closure of `E / F`
+if it is separable over `F`. -/
+theorem le_separableClosure (L : IntermediateField F E) [IsSeparable F L] :
+    L ≤ separableClosure F E := le_separableClosure' F E (IsSeparable.separable F)
+
+/-- An intermediate field of `E / F` is contained in the (relative) separable closure of `E / F`
+if and only if it is separable over `F`. -/
+theorem le_separableClosure_iff (L : IntermediateField F E) :
+    L ≤ separableClosure F E ↔ IsSeparable F L :=
+  ⟨fun h ↦ ⟨fun x ↦ by simpa only [minpoly_eq] using h x.2⟩, fun _ ↦ le_separableClosure _ _ _⟩
+
+/-- The (relative) separable closure of the (relative) separable closure of `E / F` is equal to
+itself. -/
+theorem separableClosure.separableClosure_eq_bot :
+    separableClosure (separableClosure F E) E = ⊥ := bot_unique fun x hx ↦
+  mem_bot.2 ⟨⟨x, (mem_separableClosure_iff _ E).1 hx |>.comap_minpoly_of_isSeparable F⟩, rfl⟩
```

Comment:
I'd like to add
```
theorem IsSepClosed.separableClosure_eq_bot [IsSepClosed F] : separableClosure F E = ⊥ :=
  bot_unique fun x hx ↦ mem_bot.2 <|
    have ⟨y, hy⟩ := IsSepClosed.algebraMap_surjective (⟨x, hx⟩ : separableClosure F E)
    ⟨y, Subtype.ext_iff.mp hy⟩
```
here; it would be nice to add the IsAlgClosed version as well using `integralClosure` or my `IntermediateField.algClosure` that hasn't been PR'd ... (notice that [IsIntegrallyClosed.integralClosure_eq_bot](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/IntegrallyClosed.html#IsIntegrallyClosed.integralClosure_eq_bot) says something different, because IsIntegrallyClosed is restricted to the FractionRing.)

Reply:
More generally, if `IsSepClosed F`, `L : IntermediateField F E` and `IsSeparable F L`, then `L = ⊥`. Then `separableClosure F E = ⊥` should comes automatically.

Reply:
I added `IntermediateField.eq_bot_of_isSepClosed_of_isSeparable` in `IsSepClosed.lean`. Would you like to added a similar result to `IsAlgClosed.lean`?

Reply:
> I plan to add IntermediateField.eq_bot_of_isSepClosed_of_isSeparable in IsSepClosed.lean.

Sounds good, thanks for the generalization!

> Would you like to add a similar result to IsAlgClosed.lean?

That would be nice, yes. I just think you might face some difficulties without an `IntermediateField` version of `integralClosure`.

> Oh, but separableClosure.separableClosure_eq_bot is not a corollary of it, since separableClosure F E in general is not IsSepClosed 😅

Yeah, they don't imply each other but the statements are similar, so it's a good idea to place them together. My version is used to prove `IsSepClosed.separableClosure_eq_bot_iff` below.

Reply:
Now this code works:
```lean
example [IsSepClosed F] : separableClosure F E = ⊥ := by
  exact eq_bot_of_isSepClosed_of_isSeparable _
```
but `exact?` can't find `eq_bot_of_isSepClosed_of_isSeparable `, I don't know why.